homeassistant/components/homekit/type_cameras.py

```diff
@@ -0,0 +1,250 @@
+    (640, 480),
+    (1024, 576),
+    (1024, 768),
+    (1280, 720),
+    (1280, 960),
+    (1920, 1080),
+]
+
+VIDEO_PROFILE_NAMES = ["baseline", "main", "high"]
+
+
+@TYPES.register("Camera")
+class Camera(HomeAccessory, PyhapCamera):
+    """Generate a Camera accessory."""
+
+    def __init__(self, hass, driver, name, entity_id, aid, config):
+        """Initialize a Camera accessory object."""
+        self._ffmpeg = hass.data[DATA_FFMPEG]
+        self._camera = hass.data[DOMAIN_CAMERA]
+
+        self.config = CAMERA_SCHEMA(config)
+
+        max_fps = self.config[CONF_MAX_FPS]
+        max_width = self.config[CONF_MAX_WIDTH]
+        max_height = self.config[CONF_MAX_HEIGHT]
+        resolutions = [
+            (w, h, fps)
+            for w, h, fps in SLOW_RESOLUTIONS
+            if w <= max_width and h <= max_height and fps < max_fps
+        ] + [
+            (w, h, max_fps)
+            for w, h in RESOLUTIONS
+            if w <= max_width and h <= max_height
+        ]
+
+        video_options = {
+            "codec": {
+                "profiles": [
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["BASELINE"],
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["MAIN"],
+                    VIDEO_CODEC_PARAM_PROFILE_ID_TYPES["HIGH"],
+                ],
+                "levels": [
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE3_1"],
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE3_2"],
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE4_0"],
+                ],
+            },
+            "resolutions": resolutions,
+        }
+        audio_options = {
+            "codecs": [
+                {"type": "OPUS", "samplerate": 24},
+                {"type": "AAC-eld", "samplerate": 16},
+            ]
+        }
+
+        stream_address = self.config.get(CONF_STREAM_ADDRESS) or get_local_ip()
+
+        options = {
+            "video": video_options,
+            "audio": audio_options,
+            "address": stream_address,
+        }
+
+        super().__init__(
+            hass,
+            driver,
+            name,
+            entity_id,
+            aid,
+            config,
+            category=CATEGORY_CAMERA,
+            options=options,
+        )
+
+    def update_state(self, new_state):
+        """Handle state change to update HomeKit value."""
+        pass  # pylint: disable=unnecessary-pass
+
+    def _get_stream_source(self):
+        camera = self._camera.get_entity(self.entity_id)
+        if not camera or not camera.is_on:
+            return None
+        stream_source = self.config.get(CONF_STREAM_SOURCE)
+        if stream_source:
+            return stream_source
+        try:
+            stream_source = asyncio.run_coroutine_threadsafe(
+                camera.stream_source(), self.hass.loop
+            ).result(10)
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.error("Failed to get stream source: %s", err)
```

Comment:
```
_LOGGER.exception("Failed to get stream source")
```

Would capture the full traceback.

Should this error be more actionable though? Maybe something like "Failed to get stream source - this could be a transient error or your camera might not be compatible with HomeKit yet"? Thats a bit word though..

Reply:
I think stream_source implementation is pretty much up to the integration author so maybe wordy is better in this case since almost anything can happen. I'll use that and `_LOGGER.exception` (thanks for the hint!)